### DISCUSSION TITLE
How to implement user registration without an email?

### DISCUSSION THREAD

**Discussion by Blushyes** [supabase/supabase#30617] [Questions]
What I want to achieve now is to bind a WeChat user's identifier to a Supabase Auth user after the user scans a QR code with WeChat. However, it seems that the user must verify their email?

**Comment by GaryAustin1** [comment]
You have to have an email or phone number to use Supabase auth.
Is it you don't have those, or you just don't want to do a validation step?
If you are creating a user from serverside and you know the wechat info (email, phone) is secure then you can just mark the user confirmed in the process.  This has to be done with admin auth calls on a server though.   I know of no examples for this, so just throwing out an idea.

**Reply by Blushyes** [reply]
You know, in China, most people use WeChat for third-party login instead of email, so I prefer that people can scan a QR code with WeChat to register an account. If users need to verify their email after logging in with WeChat, it might make them feel that the login process is too cumbersome and they might give up on registering (email is not used as much in China).

**Reply by Blushyes** [reply]
If Supabase Auth doesn't support binding other identity credentials like WeChat credentials, I might have to manually maintain an authentication system myself, which is obviously very troublesome. It would be great if there were a way to directly integrate Supabase Auth by binding WeChat login credentials.

**Reply by GaryAustin1** [reply]
I am not an employee and just giving you feedback on the current feature set as I know it.
If you have feed for supabase for new features there is a feature request category.
The idea I suggest would not require confirming the email if you know it is secure from the QR code to your serverside code setting up the user.  You would just mark the email as confirmed in the auth.users table.  You are taking on all risks of auth though with this process and that the emails are indeed owned by the user.

**Reply by GaryAustin1** [reply]
Also:  https://github.com/orgs/supabase/discussions/29711

**Reply by Blushyes** [reply]
> I am not an employee and just giving you feedback on the current feature set as I know it. If you have feed for supabase for new features there is a feature request category. The idea I suggest would not require confirming the email if you know it is secure from the QR code to your serverside code setting up the user. You would just mark the email as confirmed in the auth.users table. You are taking on all risks of auth though with this process and that the emails are indeed owned by the user.

Ok, thanks, I will try it.
